concordium-node/src/bin/cli.rs

```diff
@@ -660,22 +660,38 @@ async fn download_chunk(
         tempfile::NamedTempFile::new_in(data_dir_path).context("Cannot create output file.")?;
     info!("Downloading the catch-up file from {} to {}", download_url, temp_file.path().display());
     {
-        let chunk_response = http_client.get(download_url.clone()).send().await?;
-        anyhow::ensure!(
-            chunk_response.status().is_success(),
-            "Unable to download the block chunk file from {}: {} {}",
-            download_url,
-            chunk_response.status().as_str(),
-            chunk_response.status().canonical_reason().unwrap()
-        );
+        // Use a 5 minute timeout for downloading the block chunk file.
+        let timeout_duration = std::time::Duration::from_secs(300);
 
-        let file = temp_file.as_file_mut();
-        let mut buffer = std::io::BufWriter::new(file);
-        let mut stream = chunk_response.bytes_stream();
-        while let Some(Ok(bytes)) = stream.next().await {
-            buffer.write_all(&bytes)?;
-        }
-        buffer.flush()?;
+        // Download the block chunk file.
+        match tokio::time::timeout(timeout_duration, http_client.get(download_url.clone()).send())
```

Comment:
I don't think we should be using this. Reqwest supports timeouts directly

https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.connect_timeout for the connection one (which we should set to a low number of seconds)

And this https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.timeout for the request timeout. This needs to be a reasonably large, like the current timeout you set.

Reply:
I think for the second option it would be ideal to add a configuration option with a default of 5min. THat'll also make it simple to test.